rpc/dial_options.go

```diff
@@ -52,6 +53,10 @@ type dialOptions struct {
 
 	// stats monitoring on the connections.
 	statsHandler stats.Handler
+
+	// interceptors
+	unaryInterceptor  grpc.UnaryClientInterceptor
```

Comment:
can we make these lists so we can add more than one interceptor?

Reply:
it seems like we can do that by passing in multiple dial options with interceptors and then chaining those interceptors. this is the pattern we have for server options